test/test-page/test-move.vim

```diff
@@ -0,0 +1,16 @@
+source ../init.vim
+runtime plugin/wiki.vim
+
+silent edit wiki-tmp/BadName.wiki
+
+silent call wiki#page#rename('subdir/GoodName')
+call wiki#test#assert_equal(
+      \ expand('<sfile>:h') . '/wiki-tmp/subdir/GoodName.wiki',
+      \ expand('%:p'))
+
+silent edit wiki-tmp/index.wiki
+silent call wiki#page#rename('nosubdir/index')
+call wiki#test#assert_equal(
+      \ expand('<sfile>:h') . '/wiki-tmp/nosubdir/index.wiki',
+      \ expand('%:p'))
```

Comment:
And this test will fail even with your PR. For testing purposes, I think the `wiki#page#rename` function should take a "force" argument to avoid the user input, e.g. `wiki#page#rename(new_name, 1)` where the second argument is optional and if trueish implies the force.